python/cudf/cudf/core/series.py

```diff
@@ -453,31 +452,16 @@ def empty(self):
         return not len(self)
 
     def __getitem__(self, arg):
-        data = self._column[arg]
-        index = self.index.take(arg)
-        if is_scalar(data) or data is None:
-            return data
-        return self._copy_construct(data=data, index=index)
+        if isinstance(arg, (slice, range, ColumnBase)):
```

Comment:
Pandas only defers down to iloc when using slices from what I've seen. If a user passes a Column we should definitely treat that as a `loc`. Pandas doesn't support a `range` object (throws a `KeyError` exception), so we should default that to `.loc` as well if possible.

Reply:
ok, I will move them.